### ISSUE TITLE
Coachmark inconsistently shows focus outline

### ISSUE THREAD

**Issue by jendowns** [microsoft/fluentui#18273]
### Environment Information

- **Package version(s)**: `"office-ui-fabric-react": "^7.147.0"`
- **Browser and OS versions**: Edge v90, Windows 10

### Describe the issue:

⚠ Related: [Issue #153376: Fluent UI Coachmark component's teardrop is not accessible from keyboard](https://github.com/microsoft/fluentui/issues/15376)

Please watch the short video below for a demonstration (there is sound describing what I'm doing). Behavior can be reproduced on [the Coachmark example on Fluent's website](https://developer.microsoft.com/en-us/fluentui/#/controls/web/coachmark).

Right now the Coachmark's focus indicator is shown inconsistently due to (1) the way the Coachmark is activated & then programmatically gets focus, and (2) the way `data-is-focusable` applies an `-isFocusVisible` or `-isFocusHidden` class to the body of the page depending on if you used your keyboard or mouse to activate the Coachmark.

In other words:

- **If you activate the Coachmark with your mouse,** the Coachmark immediately receives focus but because you used your mouse and an `onMouseDown` event was last detected, `data-is-focusable` applies `-isFocusHidden` to the body and hides the focus outline.
- **If you activate the Coachmark with your keyboard,** the Coachmark immediately receives focus and because you used your keyboard & an `onKeyDown` even was last detected, `data-is-focusable` applies `-isFocusVisible` to the body and shows the focus outline.

https://user-images.githubusercontent.com/9057921/119060078-5eabb680-b986-11eb-8350-5f66584134b5.mp4

I realize that the purpose of `data-is-focusable` and the `onMouseDown`/`onKeyDown` event listeners is to hide the focus outline if someone is using their mouse, but this is a special situation. The Coachmark is programmatically receiving focus when activated, so the inconsistency with this outline and when it is/isn't shown has caused our accessibility testers to flag this issue in our codebase for the Coachmark. 

Another consideration: whereas [Fluent's Coachmark example](https://developer.microsoft.com/en-us/fluentui/#/controls/web/coachmark) shows the Coachmark appearing right where it's activated, it's possible that a Coachmark is activated & appears on the other side of the page. So when focus is programmatically given to a Coachmark across the page, with no focus indicator (if `onMouseDown` is detected), that could be confusing for our users.

Please let me know if you have any questions. Thank you for your help!

**Comment by gouttierre** [comment]
@jendowns - thank you for reaching out and providing details into the issue you are experiencing. 

@miroslavstastny - Would you be able to confirm if this is a regression, or if this behavior is an issue?
cc @smhigley

**Comment by ling1726** [comment]
@jendowns 
Programmatic focus natively does not create a focus-indicator, similarly to when you open a modal dialog, it will trap focus but no indicator will be visible on the first focused element.

Adding a focus indicator just for `Coachmark` would be incosistent to native behaviour and create inconsistency in the library since not every component use case might have this requirement. In this case you can take a look at this example: https://codesandbox.io/s/withered-cloud-dn6gw?file=/src/App.js

It imports the focus visible classname and you can just apply it to your coachmark so that it always displays.

@smhigley  do you think this scenario needs to be supported in Fluent ? I kind of even wonder how it was flagged as an a11y bug

**Comment by jendowns** [comment]
@ling1726 

> Programmatic focus natively does not create a focus-indicator, similarly to when you open a modal dialog, it will trap focus but no indicator will be visible on the first focused element.

I understand that and your point about adding a focus indicator. But the behavior is still inconsistent depending on how you activate the Coachmark. You will see a 1px blue focus indicator if you activate with a mouse -- is that the intention to keep this?

**Comment by smhigley** [comment]
@jendowns I'm definitely sympathetic in general to the issue of creating a divergent experience for mouse and keyboard users, and I'm not a huge fan of the idea that only keyboard users benefit from a strong focus indicator. The benefit of only showing the focus indicator on keyboard interaction is that it allows us to create a really strong, visible indicator without designers pushing back. It'd be great if a strong focus indicator were generally accepted for both mouse and keyboard users, but we're not really there in practice.

That said, in this specific scenario, the lack of a focus indicator for mouse users doesn't bother me. The visual indication of the coachmark appearing, especially with the animation is a much stronger visual cue than the 1px focus indicator for keyboard users. If anyone missed both the animated blue ring pulse + the big blue dot appearing, I doubt the thin focus rectangle is going to make any difference.

@ling1726 I agree this doesn't fall cleanly into the accessibility tag, since it's more of a general design question.

**Comment by jendowns** [comment]
@smhigley @ling1726 Thank you both! I'll take these points back to our accessibility testers.
